### PR TITLE
Split the `authentication` contract into two

### PR DESCRIPTION
**Pull request by Ereski**
Two new types: `authentication-password` and `authentication-oauth`.

Change-type: patch
Signed-off-by: Carol Schulze <carol@balena.io>

### CODE REVIEW

File: lib/backend/postgres/index.ts [hunk 1]
```diff
@@ -739,15 +775,37 @@ export class PostgresBackend implements Database {
...
+				);
+				await this.upsertObject(
+					context,
+					{
+						slug: 'link-' + userContract.id + '-' + authenticationContract.id,
 						version: '1.0.0',
 						type: 'link@1.0.0',
-						name: 'has attachment',
-						active: true,
+						name: 'is authenticated with',
```
Comment (joshbwlng) [inline lib/backend/postgres/index.ts:807]: Is this a new link name? If so, we'll need to keep it in mind for the relationship work @ramirogm is tackling.

File: lib/contracts/role-user-community.ts [hunk 2]
```diff
@@ -147,13 +147,23 @@ export const roleUserCommunity: RoleContractDefinition = {
 				},
 				{
 					properties: {
-						id: {
```
Comment (joshbwlng) [inline lib/contracts/role-user-community.ts:150]: If I'm reading the old role schema correctly, it seems like it should have never matched anything? Trying to match `authentication` or `user-settings` contracts with the same `id` as the `user`...

**Comment by joshbwlng** [comment]
@Ereski LGTM, guess all that's left are the tests like you mentioned. What do you mean by "proper permissions"? Also, do you think we will to migrate the old `authentication` contracts?

File: lib/contracts/role-user-community.ts (hunk 2 above)
Comment (Ereski) [reply lib/contracts/role-user-community.ts:150]: @joshbwlng yes lol. No idea where my head was at at the time.......

**Comment by Ereski** [comment]
@balena-ci rebase

**Comment by Ereski** [comment]
@balena-ci I self-certify!

**Pull request merged**